sinks/signalfx/signalfx.go

```diff
@@ -183,11 +201,91 @@ func NewClient(endpoint, apiKey string, client *http.Client) DPClient {
 	return httpSink
 }
 
+func MigrateConfig(conf *veneur.Config) error {
+	if conf.SignalfxAPIKey.Value != "" {
+		return nil
+	}
+	if conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod == "" {
+		conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod = "10m"
```

Comment:
I know this isn't explicitly related to the change you made, so feel free to ignore if it's not helpful - but I can't help but think it's more useful to throw a validation error here, if "dynamic per tag API key refreshing" is turned on, but this setting is not explicitly set, than it is to default a value for all instances of a SignalFx sink.

Reply:
This bit of logic is to maintain compatibility with the old format. (I added a `TODO` above to note that.) The part where I am setting the default value is
```
if config.DynamicPerTagAPIKeysRefreshPeriod == 0 {
	config.DynamicPerTagAPIKeysRefreshPeriod = time.Duration(10 * time.Minute)
}
```
below.

I updated that to error if the value is unset. This will only affect the new config format, and not the old.